standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreServerUtils.java

```diff
@@ -546,7 +552,11 @@ public static void updateTableStatsForCreateTable(Warehouse wh, Database db, Tab
       }
     }
 
-    if (MetastoreConf.getBoolVar(conf, MetastoreConf.ConfVars.STATS_AUTO_GATHER)) {
+    String statsSource = Optional.ofNullable(envContext).orElse(new EnvironmentContext(Collections.emptyMap()))
+            .getProperties().getOrDefault(HIVE_ICEBERG_STATS_SOURCE, HiveMetaHook.ICEBERG.toLowerCase()).toLowerCase();
```

Comment:
.`getOrDefault(HIVE_ICEBERG_STATS_SOURCE, HiveMetaHook.ICEBERG.toLowerCase()).toLowerCase()` could we call toLowerCase just once? :)

Reply:
````
String statsSource = Optional.ofNullable(envContext)
    .map(EnvironmentContext::getProperties)
    .map(prop -> prop.get(HIVE_ICEBERG_STATS_SOURCE))
    .orElse(HiveMetaHook.ICEBERG).toLowerCase()
````

Reply:
Done